udp_driver/src/udp_receiver_node.cpp

```diff
@@ -0,0 +1,126 @@
+// Copyright 2021 LeoDrive, Copyright 2021 the Autoware Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "udp_driver/udp_receiver_node.hpp"
+
+#include <memory>
+#include <string>
+
+namespace lc = rclcpp_lifecycle;
+using LNI = rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface;
+using lifecycle_msgs::msg::State;
+
+namespace drivers
+{
+namespace udp_driver
+{
+
+UdpReceiverNode::UdpReceiverNode(const rclcpp::NodeOptions & options)
+: lc::LifecycleNode("udp_receiver_node", options),
+  m_ctx{new IoContext(1)}
+{
+  get_params();
+}
+
+UdpReceiverNode::UdpReceiverNode(
+  const rclcpp::NodeOptions & options,
+  const std::shared_ptr<IoContext> & ctx)
+: lc::LifecycleNode("udp_receiver_node", options),
+  m_ctx{ctx}
+{
+  get_params();
+}
+
+LNI::CallbackReturn UdpReceiverNode::on_configure(const lc::State & state)
+{
+  (void)state;
+
+  m_udp_driver = std::make_unique<UdpDriver>(*m_ctx);
+
+  try {
+    m_udp_driver->init_receiver(m_ip, m_port);
+    m_udp_driver->receiver()->open();
+    m_udp_driver->receiver()->bind();
+    m_udp_driver->receiver()->asyncReceive(
+      boost::bind(&UdpReceiverNode::receiver_callback, this, _1));
+  } catch (const std::exception & ex) {
+    RCLCPP_ERROR(
```

Comment:
nice call with the error, this should throw now in case someone forgot to connect the device or something 👍🏼  do you think it would be helpful to add a `hint` or something for the user to `check their ethernet settings`? that happened to me earlier when testing when I forgot to add the IP I was trying to connect to

Reply:
@flynneva The IP should be a required field. Did it not fail to launch the node with an error like `Parameter error: expected [string] got [not set]`?

Reply:
it did fail immediately. but if I was someone completely new to this package i wouldnt have understood what the error was. it didnt say anything about "missing address" or "port" or anything specific...just that some parameter wasnt set.